src/Drivers/Parspal/Parspal.php

```diff
@@ -0,0 +1,266 @@
+use Shetabit\Multipay\Invoice;
+use Shetabit\Multipay\Receipt;
+use Shetabit\Multipay\RedirectionForm;
+use Shetabit\Multipay\Request;
+use chillerlan\SimpleCache\CacheOptions;
+use chillerlan\SimpleCache\FileCache;
+
+class Parspal extends Driver
+{
+    /**
+     * HTTP Client.
+     *
+     * @var object
+     */
+    protected $client;
+
+    /**
+     * Invoice
+     *
+     * @var Invoice
+     */
+    protected $invoice;
+
+    /**
+     * Driver settings
+     *
+     * @var object
+     */
+    protected $settings;
+
+    /**
+     * Cache
+     *
+     * @var FileCache
+     */
+    protected $cache;
+
+    protected const PAYMENT_PURCHASE_STATUS_OK = 'ACCEPTED';
+
+    protected const PAYMENT_VERIFY_STATUS_OK = 'SUCCESSFUL';
+
+    /**
+     * Constructor.
+     * Construct the class with the relevant settings.
+     *
+     * @param Invoice $invoice
+     * @param $settings
+     */
+    public function __construct(Invoice $invoice, $settings)
+    {
+        $this->invoice($invoice);
+        $this->settings = (object) $settings;
+        $this->client = new Client();
+        $this->cache = new FileCache(
+            new CacheOptions([
+                'filestorage' => $this->settings->cachePath,
+            ])
+        );
+    }
+
+    /**
+     * Retrieve data from details using its name.
+     *
+     * @return string
+     */
+    private function extractDetails($name)
+    {
+        return empty($this->invoice->getDetails()[$name]) ? null : $this->invoice->getDetails()[$name];
+    }
+
+    /**
+     * Purchase Invoice.
+     *
+     * @return string
+     *
+     * @throws PurchaseFailedException
+     */
+    public function purchase()
+    {
+        $data = [
+            'amount' => $this->getInvoiceAmount(),
+            'return_url' => $this->settings->callbackUrl,
+            'order_id' => $this->invoice->getUuid(),
+            'description' => $this->extractDetails('description'),
+            'payer' => [
+                'name' => $this->extractDetails('name'),
+                'mobile' => $this->extractDetails('mobile'),
+                'email' => $this->extractDetails('email'),
+            ],
+        ];
+
+        $response = $this->client->request(
+            'POST',
+            $this->getPurchaseUrl(),
+            [
+                'headers' => [
+                    'ApiKey' => $this->settings->merchantId,
+                    'Content-Type' => 'application/json',
+                ],
+                'json' => $data,
+                'http_errors' => false,
+            ]
+        );
+
+        $body = json_decode($response->getBody()->getContents(), true);
+
+        if ($body['status'] !== self::PAYMENT_PURCHASE_STATUS_OK) {
+            throw new PurchaseFailedException($body['message'], $body['error_code'] ?? 0);
+        }
+
+        $this->invoice->transactionId($body['payment_id']);
+
+        $this->cache->set('payment_link_' . $body['payment_id'], $body['link'], 3600);
```

Comment:
I merge this PR, but as an improvement, it would be great if you move the cache TTL (3600) to configs.

Reply:
@khanzadimahdi Thank you, I added your suggestion in another [PR](https://github.com/shetabit/multipay/pull/274).